rpc/src/v1/tests/helpers/miner_service.rs

```diff
@@ -115,22 +115,35 @@ impl MinerService for TestMinerService {
 	}
 
 	/// Imports transactions to transaction queue.
-	fn import_transactions<T>(&self, transactions: Vec<SignedTransaction>, _fetch_account: T) ->
+	fn import_transactions<T>(&self, transactions: Vec<SignedTransaction>, fetch_account: T) ->
 		Vec<Result<TransactionImportResult, Error>>
 		where T: Fn(&Address) -> AccountDetails {
 		// lets assume that all txs are valid
 		self.imported_transactions.lock().unwrap().extend_from_slice(&transactions);
 
+		for transaction in &transactions {
+			if let Ok(ref sender) = transaction.sender() {
+				let nonce = self.last_nonce(sender).unwrap_or(fetch_account(sender).nonce);
+				self.last_nonces.write().unwrap().insert(sender.clone(), nonce + U256::from(1));
```

Comment:
`let nonce = self.last_nonce(sender).map(|n| n + U256::from(1)).unwrap_or(fetch_account(sender).nonce);`

In state next valid nonce is stored, `last_nonce` returns last used nonce afair - so you should increment only in the first case.

Same thing in L144